make/autoconf/jvm-features.m4

```diff
@@ -438,6 +439,8 @@ AC_DEFUN([JVM_FEATURES_PREPARE_VARIANT],
   else
     JVM_FEATURES_VARIANT_FILTER="link-time-opt opt-size"
   fi
+  # Filter out hardlse feature by default
```

Comment:
You should not set up a PLATFORM filter in `JVM_FEATURES_PREPARE_VARIANT`. 

In fact, you will need a `JVM_FEATURES_CHECK_HARD_LSE`, which verifies that it can only be enabled on aarch64. See e.g. `JVM_FEATURES_CHECK_JVMCI` foor inspiration on how to write this and where to call it. I suggest you set up `JVM_FEATURES_PLATFORM_FILTER` in that function as well.

Reply:
Magnus, thank you for the comments. I'm taking a direction suggested by Andrew and Nick -- linux-aarch64 only solution without build configuration feature, determined by actual compiler flags. Will update the PR shortly after some final testing.